osquery/core/tables.cpp

```diff
@@ -136,6 +138,33 @@ PluginResponse TablePlugin::routeInfo() const {
   return response;
 }
 
+bool TablePlugin::isCached(size_t step) {
+  return (!FLAGS_disable_caching && step < last_cached_ + last_interval_);
+}
+
+QueryData TablePlugin::getCache() const {
+  VLOG(1) << "Retrieving results from cache for table: " << getName();
+  // Lookup results from database and deserialize.
+  std::string content;
+  getDatabaseValue(kQueries, "cache." + getName(), content);
+  QueryData results;
+  deserializeQueryDataJSON(content, results);
+  return results;
+}
+
+void TablePlugin::setCache(size_t step,
+                           size_t interval,
+                           const QueryData& results) {
+  // Serialize QueryData and save to database.
+  std::string content;
```

Comment:
This needs a check for cache enablement, otherwise it will be serializing+saving unused data.